calculator.js

```diff
@@ -0,0 +1,65 @@
+var writes = document.getElementById('screen');
+var currNumber = 0;
+var result = 0;
+
+var buttons = document.querySelector('.allButtons');
+var numberButtons = buttons.querySelectorAll('.number');
+
+for (var i = 0; i <= numberButtons.length - 1; i++) {
+
+    numberButtons[i].onclick = function() {
+
+        var buttonNumber = parseInt(this.innerHTML);
+
+        currNumber = currNumber * 10 + buttonNumber
+
+        writes.value = currNumber;
+
+    }
+};
+
+var operationButtons = buttons.querySelectorAll('.operation');
+
+for (var i = 0; i <= operationButtons.length - 1; i++) {
+    var isInitialAction = true;
+    var prevOperation;
+    operationButtons[i].onclick = function() {
+
+        var operation = this.innerHTML;
+
+ /*===================================================================*/
+        if (operation === this.innerHTML) {
```

Comment:
This IF does not make sense, since you are already assigning innerHTML to operation the line above. Just remove the entire IF.